### PR TITLE
Give error message in DMs on hierarchy error

### LINKED ISSUE #114: Handle errors in Menu reacts

### PR DESCRIPTION
**Pull request by 1Revenger1**
Closes #114

### CODE REVIEW

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java [hunk 1]
```diff
@@ -22,36 +23,54 @@ public class ReactionProcessing extends ListenerAdapter {
...
-                guild.addRoleToMember(user, role).complete();
-            } catch (ErrorResponseException e) {
-                if (e.getErrorResponse() == ErrorResponse.UNKNOWN_ROLE) {
-                    removeRole(event, role);
-                }
-            } catch (HierarchyException e) {
-                System.err.println("Role is higher than Bot role");
+        if (role == null) {
+            return;
+        }
```
Comment (joshuabrown419) [inline src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:28]: one line if?

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java [hunk 2]
```diff
@@ -22,36 +23,54 @@ public class ReactionProcessing extends ListenerAdapter {
...
+
+        Member user = event.retrieveMember().complete();
+        Guild guild = event.getGuild();
+        try {
+            guild.addRoleToMember(user, role).complete();
+        } catch (ErrorResponseException e) {
+            if (e.getErrorResponse() == ErrorResponse.UNKNOWN_ROLE) {
+                removeRole(event, role);
             }
```
Comment (joshuabrown419) [inline src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:37]: Should probably also send a message to the user in the case they try to add a role that no longer exists so they know what happened

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java [hunk 3]
```diff
@@ -22,36 +23,54 @@ public class ReactionProcessing extends ListenerAdapter {
...
+        } catch (HierarchyException e) {
+            // Attempt to DM user that an error occurred
+            user.getUser().openPrivateChannel().queue((var thread) ->
+                thread.sendMessage(
+                    "Was unable to remove the role `@" + role.getAsMention() + "` from `" + guild.getName() + "`. " +
+                    "This is because `@" + role.getAsMention() + "` is above me in the role list. `" +
+                    "Please contact an admin within `" + guild.getName() + "` server to fix this."
+                ).queue(null, new ErrorHandler().ignore(ErrorResponse.CANNOT_SEND_TO_USER))
+            );
         }
```
Comment (joshuabrown419) [inline src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:74]: Any reason to not move this code into a method so its not duplicated? We could just replace assign/remove with literally `assign/remove` or just `modify`

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java [hunk 4]
```diff
@@ -22,36 +23,54 @@ public class ReactionProcessing extends ListenerAdapter {
...
+            if (e.getErrorResponse() == ErrorResponse.UNKNOWN_ROLE) {
+                removeRole(event, role);
             }
+        } catch (HierarchyException e) {
+            // Attempt to DM user that an error occurred
+            user.getUser().openPrivateChannel().queue((var thread) ->
+                thread.sendMessage(
+                    "Was unable to assign the role `@" + role.getAsMention() + "` from `" + guild.getName() + "`. " +
+                    "This is because `@" + role.getAsMention() + "` is above me in the role list. " +
+                    "Please contact an admin within `" + guild.getName() + "` server to fix this."
```
Comment (joshuabrown419) [inline src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:44]: Thought we were gonna change this to be `from` instead of `within`

**Review by joshuabrown419 — CHANGES_REQUESTED** [review]
Duplicated code probably maybe bad

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java (hunk 1 above)
Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:28]: Something wrong with this?

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java (hunk 4 above)
Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:44]: Oh, you were talking about the last line? I thought you were talking about the first line, oops. Of makes way more sense there lol.

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java (hunk 3 above)
Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:74]: Uhhh, I didn't think about doing that. They were next to each other so I guess it wasn't too egregious when I was writing it. I can make it a method though.

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java (hunk 1 above)
Comment (joshuabrown419) [reply src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:28]: I mean as in make it a oneliner
`if (role == null) return;`

Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:28]: Oh. Is that a style we want? ~~MISRA standards require brackets in our embedded stuff so I e sort of formed that habit~~

Comment (joshuabrown419) [reply src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:28]: if you want to leave it as is its fine, looks cleaner to make it one line imo

**Review by joshuabrown419 — APPROVED** [review]
LGTM

**Pull request merged**